STM32N6: Add display support

hfruchet-st opened the pull request:
Add display support for STM32N6. Enabled on STM32N6570-DK board running Rockchip
RK050HR18C 800x480 RGB 24bits panel connected to LTDC display parallel interface.
Framebuffer is configured in 800x480 RGB565 16 bits.

Tested on STM32N6570-DK:
west build -b stm32n6570_dk samples/drivers/display

Non-regression made on STM32H750B-DK:
west build -b stm32h750b_dk samples/drivers/display

hfruchet-st committed:
dts: arm: st: n6: add ltdc node

Add LTDC node for STM32N6.

Signed-off-by: Hugues Fruchet <hugues.fruchet@foss.st.com>

hfruchet-st committed:
drivers: display: stm32_ltdc: add support of clock_configure

Add support of clock_configure() for clock source selection
through devicetree.

Signed-off-by: Hugues Fruchet <hugues.fruchet@foss.st.com>

hfruchet-st committed:
drivers: memc: stm32 xspi: add psram linker section

Add stm32_psram PSRAM linker section.

Signed-off-by: Hugues Fruchet <hugues.fruchet@foss.st.com>

hfruchet-st committed:
drivers: display: stm32_ltdc: add support of framebuffer in psram

Add support of framebuffer in PSRAM memory.

Signed-off-by: Hugues Fruchet <hugues.fruchet@foss.st.com>

hfruchet-st committed:
drivers: display: stm32_ltdc: configure RIF for LTDC layer 1

Configure RIF for LTDC layer 1.

Signed-off-by: Hugues Fruchet <hugues.fruchet@foss.st.com>

hfruchet-st committed:
boards: st: stm32n6570_dk: add display support

Add support of Rockchip RK050HR18C 800x480 RGB 24bits
panel connected on LTDC display parallel interface of
STM32N6570-DK board.

Signed-off-by: Hugues Fruchet <hugues.fruchet@foss.st.com>

Commit:
samples: drivers: display: add stm32n6570_dk specific conf

Add stm32n6570_dk conf file in order to increase the amount of
HEAP to ensure k_malloc allocation goes well.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

JarmouniA reviewed (changes requested):
Welcome to Zephyr :)

hfr72 commented on boards/st/stm32n6570_dk/stm32n6570_dk_common.dtsi:
MB1860 is not sold separately by ST as it is done for MB1166 and the STM32N6570-DK has no optional display support: display panel is part of STM32N6570-DK, so we don't expose that through a shield.

gautierg-st reviewed (changes requested):
Thanks for your great work, I only have a few comments

hfruchet-st committed:
drivers: clock: stm32: h7: fixed domain clock configuration

In some case, we may need to describe a domain clock for a device
while there is no way to configure it.

Signed-off-by: Hugues Fruchet <hugues.fruchet@foss.st.com>
